kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -419,192 +401,180 @@ private LogSegment constructLogSegmentFromFileList(
       }
     }
 
-    logDebug(
-        () ->
-            format(
-                "newFiles: %s",
-                Arrays.toString(
-                    newFiles.stream().map(x -> new Path(x.getPath()).getName()).toArray())));
+    logDebugFileStatuses("listedFileStatuses", listedFileStatuses);
 
-    Tuple2<List<FileStatus>, List<FileStatus>> checkpointsAndDeltas =
+    ////////////////////////////////////////////////////////////////////////////
+    // Step 5: Partition $listedFileStatuses into the checkpoints and deltas. //
+    ////////////////////////////////////////////////////////////////////////////
+
+    final Tuple2<List<FileStatus>, List<FileStatus>> listedCheckpointAndDeltaFileStatuses =
         ListUtils.partition(
-            newFiles,
+            listedFileStatuses,
             fileStatus -> FileNames.isCheckpointFile(new Path(fileStatus.getPath()).getName()));
-    final List<FileStatus> checkpoints = checkpointsAndDeltas._1;
-    final List<FileStatus> deltas = checkpointsAndDeltas._2;
-
-    logDebug(
-        () ->
-            format(
-                "\ncheckpoints: %s\ndeltas: %s",
-                Arrays.toString(
-                    checkpoints.stream().map(x -> new Path(x.getPath()).getName()).toArray()),
-                Arrays.toString(
-                    deltas.stream().map(x -> new Path(x.getPath()).getName()).toArray())));
-
-    // Find the latest checkpoint in the listing that is not older than the versionToLoad
-    final CheckpointInstance maxCheckpoint =
-        versionToLoadOpt.map(CheckpointInstance::new).orElse(CheckpointInstance.MAX_VALUE);
-    logger.debug("lastCheckpoint: {}", maxCheckpoint);
+    final List<FileStatus> listedCheckpointFileStatuses = listedCheckpointAndDeltaFileStatuses._1;
```

Comment:
nit: I don't think we need the `listed` prefix for all these variables (these plus those above); seems clear enough and can shorten them all a bit

Reply:
I actually disagree. There are the checkpoint file statuses we have `listed` and then there's the latest+complete checkpoint file statuses that we end up selecting. I think emphasizing that these are the listed ones -- but that doesn't mean they are the ones we are selecting to be in the log segmnt, is important.

See below:

```java
////////////////////////////////////////////////////////////////////////////////////////////
// Step 10: Grab the actual checkpoint file statuses for latestCompleteCheckpointVersion. //
////////////////////////////////////////////////////////////////////////////////////////////

final List<FileStatus> latestCompleteCheckpointFileStatuses =
```